Libraries/LibWeb/WebAudio/ChannelSplitterNode.cpp

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2025, Tim Ledbetter <tim.ledbetter@ladybird.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibWeb/WebAudio/BaseAudioContext.h>
+#include <LibWeb/WebAudio/ChannelSplitterNode.h>
+
+namespace Web::WebAudio {
+
+GC_DEFINE_ALLOCATOR(ChannelSplitterNode);
+
+ChannelSplitterNode::ChannelSplitterNode(JS::Realm& realm, GC::Ref<BaseAudioContext> context, ChannelSplitterOptions const& options)
+    : AudioNode(realm, context)
+    , m_number_of_outputs(options.number_of_outputs)
+{
+}
+
+ChannelSplitterNode::~ChannelSplitterNode() = default;
+
+WebIDL::ExceptionOr<GC::Ref<ChannelSplitterNode>> ChannelSplitterNode::create(JS::Realm& realm, GC::Ref<BaseAudioContext> context, ChannelSplitterOptions const& options)
+{
+    return construct_impl(realm, context, options);
+}
+
+WebIDL::ExceptionOr<GC::Ref<ChannelSplitterNode>> ChannelSplitterNode::construct_impl(JS::Realm& realm, GC::Ref<BaseAudioContext> context, ChannelSplitterOptions const& options)
+{
+    // https://webaudio.github.io/web-audio-api/#dom-baseaudiocontext-createchannelsplitter
+    // An IndexSizeError exception MUST be thrown if numberOfOutputs is less than 1 or is greater than the number of supported channels.
+    if (options.number_of_outputs < 1 || options.number_of_outputs > BaseAudioContext::MAX_NUMBER_OF_CHANNELS)
+        return WebIDL::IndexSizeError::create(realm, "Invalid number of outputs"_string);
+
+    // https://webaudio.github.io/web-audio-api/#audionode-channelcountmode-constraints
+    // The channel count mode cannot be changed from "explicit" and an InvalidStateError exception MUST be thrown for any attempt to change the value.
+    if (options.channel_count_mode.has_value() && options.channel_count_mode != Bindings::ChannelCountMode::Explicit)
```

Comment:
This should be implemented in an override of `set_channel_count_mode` so that it also applies outside of the constructor. Similar for all of the the other exception checks that we need to perform here - initialize_audio_node_options is meant to throw the exception here.